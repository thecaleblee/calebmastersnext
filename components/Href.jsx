import React from "react";
import Link from "next/link";
import { Link as Scroller, animateScroll as scroll } from "react-scroll";

export default function Href(props) {
  /**
   * @param {URL} background - URL of an image location
   * @param {string} href - The href of the link
   * @param {string} al - Aria-Label for the link
   * @param {string} addClass - Class names to add to link
   * @param {bool} samePage - Enable to make smooth scrolling anchor links
   * @param {string} rel - Use to include 'noopener' on links
   *
   * @example <Href background={optional: URL-to-background} href={href-of-link} al={aria-label for link} external addClass={optional class names} samePage />
   */
  const {
    samePage = false,
    background = "",
    href = "/",
    al = "",
    addClass = "",
    children,
    tabindex = 0,
    rel = "",
  } = props;

  return (
    <>
      {samePage ? (
        <Scroller
          activeClass="active"
          to={href}
          spy={false}
          smooth={true}
          offset={0}
          duration={500}
          aria-label={al}
          className={addClass}
        >
          {children}
        </Scroller>
      ) : (
        <>
          <style jsx>{`
            a:before {
              background: url(${background}) no-repeat center center;
              background-size: cover;
            }
          `}</style>
          <Link href={href}>
            {addClass ? (
              <a
                href={href}
                aria-label={al}
                className={addClass}
                tabIndex={tabindex}
                rel={rel}
              >
                {children}
              </a>
            ) : (
              <a href={href} aria-label={al} rel={rel} tabIndex={tabindex}>
                {children}
              </a>
            )}
          </Link>
        </>
      )}
    </>
  );
}
