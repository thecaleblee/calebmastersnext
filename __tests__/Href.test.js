import React from "react";
import { describe, it } from "@jest/globals";
import { shallow } from "enzyme";
import Href from "../components/Href";

describe("<Href />", () => {
  it("Should render without props", () => {
    const wrapper = shallow(<Href />);

    expect(wrapper.contains(<a aria-label="" href="/" />)).toBe(true);
  });
  it("Should update the href to '/tests'", () => {
    const wrapper = shallow(<Href href={`/test`} />);

    expect(wrapper.contains(<a aria-label="" href="/test" />)).toBe(true);
  });
  it("Should update the Aria Label to 'Test Link'", () => {
    const wrapper = shallow(<Href al={`Test Link`} />);

    expect(wrapper.contains(<a aria-label="Test Link" href="/" />)).toBe(true);
  });
  it("Should include the classes'test class'", () => {
    const wrapper = shallow(<Href addClass={`test class`} />);

    expect(wrapper.find(".test").length).toBe(1);
  });
  it("Should switch to link anchoring when 'samePage' is passed in", () => {
    const wrapper = shallow(<Href samePage />);

    expect(wrapper.contains(<a aria-label="" href="/test" />)).toBe(false);
    expect(wrapper.children().props()).toHaveProperty("smooth", true);
  });
  it("Snapshot of normal link should match", () => {
    const component = shallow(<Href samePage={false} />);
    expect(component).toMatchSnapshot();
  });
  it("Snapshot of samePage link should match", () => {
    const component = shallow(<Href samePage={true} />);
    expect(component).toMatchSnapshot();
  });
});
