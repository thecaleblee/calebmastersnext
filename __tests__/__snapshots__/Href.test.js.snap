// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Href /> Snapshot of normal link should match 1`] = `
<Fragment>
  <style
    jsx={true}
  >
    
            a:before {
              background: url() no-repeat center center;
              background-size: cover;
            }
          
  </style>
  <Link
    href="/"
  >
    <a
      aria-label=""
      href="/"
    />
  </Link>
</Fragment>
`;

exports[`<Href /> Snapshot of samePage link should match 1`] = `
<Fragment>
  <Link
    activeClass="active"
    aria-label=""
    className=""
    duration={500}
    offset={0}
    smooth={true}
    spy={false}
    to="/"
  />
</Fragment>
`;
